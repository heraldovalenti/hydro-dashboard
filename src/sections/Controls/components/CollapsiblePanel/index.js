import {
  ExpansionPanel,
  ExpansionPanelDetails,
  ExpansionPanelSummary,
} from '@material-ui/core';
import { withStyles } from '@material-ui/core/styles';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
import React, { useState } from 'react';

const StyledExpansionPanel = withStyles({
  root: {
    borderRadius: '4px',
    border: '1px solid #d5d5d5',
    boxShadow: '0 2px 3px rgba(0, 0, 0, 0.05)',
  },
})(ExpansionPanel);

const StyledExpansionPanelSummary = withStyles({
  content: {
    margin: '0 !important',
    minHeight: 'unset !important',
  },
  expanded: {
    margin: '0 !important',
    minHeight: 'unset !important',
  },
})(ExpansionPanelSummary);

const StyledExpansionPanelDetails = withStyles({
  root: {
    padding: '16px',
    paddingTop: 0,
  },
})(ExpansionPanelDetails);

export default function CollapsiblePanel({
  expanded = false,
  title,
  children,
}) {
  const [isExpanded, setIsExpanded] = useState(expanded);

  return (
    <StyledExpansionPanel
      className="collapsible--panel"
      expanded={isExpanded}
      onChange={() => setIsExpanded(!isExpanded)}
    >
      <StyledExpansionPanelSummary expandIcon={<ExpandMoreIcon />}>
        <h4>{title}</h4>
      </StyledExpansionPanelSummary>
      <StyledExpansionPanelDetails>
        <div className="collapsible--panel__content">{children}</div>
      </StyledExpansionPanelDetails>
    </StyledExpansionPanel>
  );
}
