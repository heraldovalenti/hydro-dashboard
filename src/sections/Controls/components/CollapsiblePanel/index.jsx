import { Accordion, AccordionDetails, AccordionSummary } from '@mui/material';
import { withStyles } from '@mui/styles';
import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
import React, { useState } from 'react';

const StyledAccordion = withStyles({
  root: {
    borderRadius: '4px',
    border: '1px solid #d5d5d5',
    boxShadow: '0 2px 3px rgba(0, 0, 0, 0.05)',
  },
})(Accordion);

const StyledAccordionSummary = withStyles({
  content: {
    margin: '0 !important',
    minHeight: 'unset !important',
  },
  expanded: {
    margin: '0 !important',
    minHeight: 'unset !important',
  },
})(AccordionSummary);

const StyledAccordionDetails = withStyles({
  root: {
    padding: '16px',
    paddingTop: 0,
  },
})(AccordionDetails);

export default function CollapsiblePanel({
  expanded = false,
  title,
  children,
}) {
  const [isExpanded, setIsExpanded] = useState(expanded);

  return (
    <StyledAccordion
      className="collapsible--panel"
      expanded={isExpanded}
      onChange={() => setIsExpanded(!isExpanded)}
    >
      <StyledAccordionSummary expandIcon={<ExpandMoreIcon />}>
        <h4>{title}</h4>
      </StyledAccordionSummary>
      <StyledAccordionDetails>
        <div className="collapsible--panel__content">{children}</div>
      </StyledAccordionDetails>
    </StyledAccordion>
  );
}
