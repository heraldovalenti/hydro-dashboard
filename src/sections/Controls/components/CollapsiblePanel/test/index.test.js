import React from 'react';
import CollapsiblePanel from '../';
import ShallowRenderer from 'react-test-renderer/shallow';

describe('<CollapsiblePanel/>', () => {
  it('renders default', () => {
    const testProps = { expanded: false, title: 'Hi Im an expanded title' };
    const renderer = new ShallowRenderer();
    renderer.render(
      <CollapsiblePanel {...testProps}>
        <div>Im the content</div>
      </CollapsiblePanel>
    );
    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });
});
