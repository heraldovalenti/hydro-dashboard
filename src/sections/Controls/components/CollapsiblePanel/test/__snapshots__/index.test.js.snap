// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CollapsiblePanel/> renders default 1`] = `
<ForwardRef(WithStyles)
  className="collapsible--panel"
  expanded={false}
  onChange={[Function]}
>
  <ForwardRef(WithStyles)
    expandIcon={<UNDEFINED />}
  >
    <h4>
      Hi Im an expanded title
    </h4>
  </ForwardRef(WithStyles)>
  <ForwardRef(WithStyles)>
    <div
      className="collapsible--panel__content"
    >
      <div>
        Im the content
      </div>
    </div>
  </ForwardRef(WithStyles)>
</ForwardRef(WithStyles)>
`;
